cmd/sonobuoy/app/worker.go

```diff
@@ -107,15 +124,15 @@ func runGatherSingleNode(cmd *cobra.Command, args []string) {
 	// http://sonobuoy-master:8080/api/v1/results/by-node/node1/systemd_logs
 	url := cfg.MasterURL + "/" + cfg.NodeName + "/" + cfg.ResultType
 
-	err = worker.GatherResults(cfg.ResultsDir+"/done", url, client)
+	err = worker.GatherResults(cfg.ResultsDir+"/done", url, client, stop)
 	if err != nil {
 		errlog.LogError(err)
 		os.Exit(1)
 	}
-
 }
 
 func runGatherGlobal(cmd *cobra.Command, args []string) {
+	stop := sigHandler()
```

Comment:
should I inline this into 152?

Reply:
If it's only used inside of GatherResults, then I don't see why it needs to be passed in.  From what I see that looks like the only place.

Reply:
my logic there was to make it obvious from the start that signal handling was happening, but i'll move it deeper and see what it looks like